### PR TITLE
Compile updated messages

### PR DESCRIPTION
**Pull request by rlucioni**
When .po files are updated, they need to be compiled to .mo files for use with gettext. This is a project-agnostic approach for accomplishing that.

@edx/ecommerce FYI. This will prevent issues like the one detailed at https://github.com/edx/course-discovery/pull/499#issuecomment-269513662, and has already been run against programs (https://github.com/edx/programs/pull/190).

### CODE REVIEW

File: transifex/pull.py [hunk 1]
```diff
@@ -103,8 +104,8 @@ def branch(self):
         """Create and check out a new branch."""
         subprocess.run(['git', 'checkout', '-b', self.branch_name], check=True)
 
-    def pull(self):
-        """Download translated strings from Transifex.
+    def update_translations(self):
```
Comment (mikedikan) [inline transifex/pull.py]: why the change

Comment (rlucioni) [inline transifex/pull.py]: Readability and clarity. This is no longer just pulling updated messages from Transifex.
